### ISSUE TITLE
Improve `dict-init-mutate` msg suggestion

### ISSUE THREAD

**Issue by clavedeluna** [PyCQA/pylint#7819]
### Current problem

This [PR](https://github.com/PyCQA/pylint/pull/7794) added the `dict-init-mutate` check with a basic msg of `Declare all known key/values when initializing the dictionary.`. This issue is to improve that message to read something more like `Declare all known key/values when initializing the dictionary with 'config = {'dir': 'bin', 'user': 'me', 'workers': 5}'` which will require building the dictionary from nodes.

### Desired solution

Use suggestion provided here https://github.com/PyCQA/pylint/pull/7794/files#r1027348441 to add code to `DictInitMutateChecker` to improve the msg suggestion.

**Comment by Pierre-Sassoulas** [comment]
An interesting test case:
```python
        d = {"name": "table", "expectedrows": expectedrows}
        d["description"] = {a.cname: a.typ for a in self.axes}
```        
Not sure of the message we should show when there is already some keys defined in the constructor.
